topology/topology.h

```diff
@@ -89,6 +90,23 @@ class LayerMetadata : public NodeCollectionMetadata
     return first_node_id_;
   }
 
+  void slice( size_t start, size_t stop, size_t step, NodeCollectionPTR node_collection );
```

Comment:
Could you add a docstring for `slice`, explaining what it does?

Reply:
Done! But in the parent class, `NodeCollectionMetadata` in NodeCollection.h.